parm/archive/master_gdas.yaml.j2

```diff
@@ -5,99 +5,106 @@
-{% include "gdasice.yaml.j2" %}
-{% endfilter %}
+    # Ice data
+    {% filter indent(width=4) %}
+        {%- include "gdasice.yaml.j2" %}
+    {% endfilter %}
 {% endif %}
 
 {% if DO_OCN %}
-# Ocean forecast products
-{% filter indent(width=4) %}
-{% include "gdasocean.yaml.j2" %}
-{% endfilter %}
-{% if DO_JEDIOCNVAR and MODE == "cycled" %}
-# Ocean analysis products
-{% filter indent(width=4) %}
-{% include "gdasocean_analysis.yaml.j2" %}
-{% endfilter %}
-{% endif %}
+    # Ocean forecast products
+    {% filter indent(width=4) %}
+        {%- include "gdasocean.yaml.j2" %}
+    {% endfilter %}
+    {% if DO_JEDIOCNVAR and MODE == "cycled" %}
+        # Ocean analysis products
+        {% filter indent(width=4) %}
+            {%- include "gdasocean_analysis.yaml.j2" %}
+        {% endfilter %}
+    {% endif %}
 {% endif %}
 
 {% if DO_WAVE %}
-# Wave products
-{% filter indent(width=4) %}
-{% include "gdaswave.yaml.j2" %}
-{% endfilter %}
+    # Wave products
+    {% filter indent(width=4) %}
+        {%- include "gdaswave.yaml.j2" %}
+    {% endfilter %}
 {% endif %}
 
 {% if MODE == "cycled" %}
-# Determine if we will save restart ICs or not (only valid for cycled)
-{% set save_warm_start_forecast, save_warm_start_cycled = ( False, False ) %}
+    # Determine if we will save restart ICs or not (only valid for cycled)
+    {% set save_warm_start_forecast, save_warm_start_cycled = ( False, False ) %}
 
-{% if ARCH_CYC == cycle_HH | int%}
-# Save the warm and forecast-only cycle ICs every ARCH_WARMICFREQ days
-{% if (current_cycle - SDATE).days % ARCH_WARMICFREQ == 0 %}
-{% set save_warm_start_forecast = True %}
-{% set save_warm_start_cycled = True %}
-# Save the forecast-only restarts every ARCH_FCSTICFREQ days
-{% elif (current_cycle - SDATE).days % ARCH_FCSTICFREQ == 0 %}
-{% set save_warm_start_forecast = True %}
-{% endif %}
-{% endif %}
+    {% if ARCH_CYC == cycle_HH | int%}
+        # Save the warm and forecast-only cycle ICs every ARCH_WARMICFREQ days
+        {% if (current_cycle - SDATE).days % ARCH_WARMICFREQ == 0 %}
+            {% set save_warm_start_forecast = True %}
+            # Save the forecast-only restarts every ARCH_FCSTICFREQ days
+        {% elif (current_cycle - SDATE).days % ARCH_FCSTICFREQ == 0 %}
+            {% set save_warm_start_forecast = True %}
+        {% endif %}
 
-{% if save_warm_start_forecast %}
-# Save warm start forecast-only data
-# Atmosphere restarts
-{% filter indent(width=4) %}
-{% include "gdas_restarta.yaml.j2" %}
-{% endfilter %}
+        # The GDAS ICs (group b) are restarts and always lag increments by assim_freq
+        {% set ics_offset = (assim_freq | string + "H") | to_timedelta %}
+        {% if (current_cycle | add_to_datetime(ics_offset) - SDATE).days % ARCH_WARMICFREQ == 0 %}
+            {% if (ARCH_CYC - assim_freq) % 24 == cycle_HH | int %}
+                {% set save_warm_start_cycled = True %}
+            {% endif %}
+        {% endif %}
+    {% endif %}
```

Comment:
@TravisElless-NOAA Here are the actual changes to selecting when to archive restartb datasets.